[Detection Engine][Docs] Added response and request descriptions for API docs

yctercero opened the pull request:
# Summary

As part of the effort to add missing content for Security APIs, this PR introduces a few missing request, response, and parameter examples for Detection Engine Exception APIs. 

# How to Test

Run a local preview in bump, which shows what the output will look like in our production. 

- Install the [bump CLI](https://docs.bump.sh/help/continuous-integration/cli/) 
- `cd x-pack/solutions/security/packages/kbn-securitysolution-exceptions-common` and then run `npm run openapi:generate` followed by `npm run openapi:bundle`. That last command gives you the path to the bundled file at the end that is used in the next step.
- Run `bump preview x-pack/solutions/security/packages/kbn-securitysolution-exceptions-common/docs/openapi/ess/security_solution_exceptions_api_2023_10_31.bundled.schema.yaml`

<!--ONMERGE {"backportTargets":["8.17","8.x"]} ONMERGE-->

yctercero committed:
Added response and request descriptions per what was identified as missing in docs sheet

elasticmachine commented:
Pinging @elastic/security-detection-engine (Team:Detection Engine)

kibanamachine committed:
[CI] Auto-commit changed files from 'make api-docs'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/264971)
* Commit: 917d27b4e026621e94d36571b6ce81ed9ca92aac

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/264971#019442e0-f950-4848-b258-d44c6e274e69)

cc @yctercero

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264971","number":264971,"commit":"917d27b4e026621e94d36571b6ce81ed9ca92aac"}],"number":264971}
buildkite-pr-comment-->

yctercero committed:
Delete packages/kbn-repo-packages/package-map.json - unsure why that was created

yctercero committed:
Merge branch 'main' into update_exception_list_openapi_info

vitaliidm commented on x-pack/solutions/security/packages/kbn-securitysolution-exceptions-common/api/create_exception_list/create_exception_list.schema.yaml:
```diff
@@ -20,6 +20,14 @@ paths:
           application/json:
             schema:
               type: object
```
when I tried to execute example, I got a following error

<img width="1143" alt="Screenshot 2025-01-08 at 11 15 03" src="https://github.com/user-attachments/assets/1ffc0dd1-d280-4ad3-b347-9c3dc20d259a" />


Is part of this also to verify APIs work correctly?

yctercero replied:
This is a great catch. I'll follow up on this with the docs team. cc @jmikell821

yctercero replied:
This is not a part of our Bump docs it seems. So we're ok here.

vitaliidm reviewed (commented):
Thanks for working on this PR, @yctercero 

I left a few comments.

Also, I noticed there are changes that enable/disable tests. Are these changes supposed to be in this PR?

yctercero commented on x-pack/test/security_solution_api_integration/test_suites/lists_and_exception_lists/authorization/exceptions/lists/essentials_tier/index.ts:
```diff
@@ -8,17 +8,17 @@ import { FtrProviderContext } from '../../../../../../ftr_provider_context';
+    // loadTestFile(require.resolve('./endpoint_operations_analyst'));
+    // loadTestFile(require.resolve('./endpoint_policy_manager'));
+    // loadTestFile(require.resolve('./platform_engineer'));
+    // loadTestFile(require.resolve('./editor'));
```
😨 that got brought in from other work I was doing. Thank you for catching. I'll re-review what was committed here.

yctercero commented on x-pack/solutions/security/plugins/security_solution/jest.config.dev.js:
```diff
@@ -9,7 +9,7 @@ module.exports = {
   rootDir: '../../../../../',
   projects: [
-    '<rootDir>/x-pack/solutions/security/plugins/security_solution/common/*/jest.config.js',
+    // '<rootDir>/x-pack/solutions/security/plugins/security_solution/common/*/jest.config.js',
```
😨 that got brought in from other work I was doing. Thank you for catching. I'll re-review what was committed here.

yctercero committed:
removing files from a separate task

yctercero committed:
Merge branch 'main' of github.com:elastic/kibana into update_exception_list_openapi_info

yctercero committed:
removing files from a separate task

kibanamachine committed:
[CI] Auto-commit changed files from 'yarn openapi:generate'

yctercero committed:
addressing pr feedback and adding more examples

yctercero committed:
addressing pr feedback and adding more examples

yctercero committed:
Merge branch 'update_exception_list_openapi_info' of github.com:yctercero/kibana into update_exception_list_openapi_info

yctercero committed:
Merge branch 'main' of github.com:elastic/kibana into update_exception_list_openapi_info

yctercero committed:
removing unnecessary changes

yctercero commented on x-pack/solutions/security/packages/kbn-securitysolution-exceptions-common/api/create_exception_list/create_exception_list.schema.yaml:
```diff
@@ -20,6 +20,14 @@ paths:
           application/json:
```
I'm trying to figure out whether we need to delete this one. I couldn't find this route anywhere in our code. If anything will follow up.

yctercero replied:
I updated to add info for this route as we decided not to remove.

yctercero commented:
Thank you for the review @vitaliidm ! I've updated with further examples and removed unnecessary changes you caught.

kibanamachine committed:
[CI] Auto-commit changed files from 'make api-docs'

kibanamachine committed:
[CI] Auto-commit changed files from 'yarn openapi:generate'

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/265773)
* Commit: 0be332dd5139a45d4f8ceb654936ba2746d3bb72

### Failed CI Steps
* [FTR Configs #81](https://buildkite.com/elastic/kibana-pull-request/builds/265773#01944c65-3228-45f8-8cdb-172ac9dabc60)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/265773#01944c65-3228-45f8-8cdb-172ac9dabc60) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/265773/jobs/01944c65-3228-45f8-8cdb-172ac9dabc60/artifacts/01944c7f-4bcb-4214-acf2-9bd68f162056) FTR Configs #<span></span>81 / Serverless Common UI - Management Search bar features list features has the correct features enabled

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #264994](https://buildkite.com/elastic/kibana-pull-request/builds/264994) failed bf44ad19994cce68c32caab930e748fe71985883
* :broken_heart: [Build #264971](https://buildkite.com/elastic/kibana-pull-request/builds/264971) failed 917d27b4e026621e94d36571b6ce81ed9ca92aac

cc @yctercero

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265773","number":265773,"commit":"0be332dd5139a45d4f8ceb654936ba2746d3bb72"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264994","number":264994,"commit":"bf44ad19994cce68c32caab930e748fe71985883"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264971","number":264971,"commit":"917d27b4e026621e94d36571b6ce81ed9ca92aac"}],"number":265773}
buildkite-pr-comment-->

natasha-moore-elastic reviewed (commented):
Thanks for adding these, @yctercero! Will the missing parameter descriptions (column B in [this spreadsheet](https://docs.google.com/spreadsheets/d/1WnU-XaE8WIL1byaZCThy5aENF00uVvvlIYwzSkeiKtw/edit?gid=448060195#gid=448060195)) be added separately?

yctercero commented:
> Thanks for adding these, @yctercero! Will the missing parameter descriptions (column B in [this spreadsheet](https://docs.google.com/spreadsheets/d/1WnU-XaE8WIL1byaZCThy5aENF00uVvvlIYwzSkeiKtw/edit?gid=448060195#gid=448060195)) be added separately?

I'm having a hard time figuring out how to do this - we just reference out to common schemas which means we can't add a description as the description will differ for each person utilizing it. If you have a good example of how to do this, let me know. I'm just ramping up on OpenAPI.

yctercero commented on x-pack/solutions/security/packages/kbn-securitysolution-exceptions-common/api/read_exception_list/read_exception_list.schema.yaml:
```diff
@@ -17,25 +17,47 @@ paths:
           description: Either `id` or `list_id` must be specified
           schema:
             $ref: '../model/exception_list_common.schema.yaml#/components/schemas/ExceptionListId'
+          example: 9e5fc75a-a3da-46c5-96e3-a2ec59c6bb85
```
<img width="1723" alt="Screenshot 2025-01-13 at 11 39 05 AM" src="https://github.com/user-attachments/assets/d9de4e53-c429-41de-90f4-03c58ece0326" />

It looks like with this new version, you can select the different params and execute it to get the example curl request with the params. Not sure if that's how it'll show in our production. Just using the Swagger editor to test out.

natasha-moore-elastic replied:
You can actually run a local preview in bump, which shows what the output will look like in our production. You'd just need to install the [bump CLI](https://docs.bump.sh/help/continuous-integration/cli/) and run the [preview](https://docs.bump.sh/help/continuous-integration/cli/#bump-preview-file) command on the bundled file, for example:
`bump preview x-pack/solutions/security/packages/kbn-securitysolution-exceptions-common/docs/openapi/ess/security_solution_exceptions_api_2023_10_31.bundled.schema.yaml`. (one caveat is that the preview expires after 30 min)

yctercero replied:
See comment here - https://github.com/elastic/kibana/pull/205822#issuecomment-2591682113

natasha-moore-elastic commented:
> > Thanks for adding these, @yctercero! Will the missing parameter descriptions (column B in [this spreadsheet](https://docs.google.com/spreadsheets/d/1WnU-XaE8WIL1byaZCThy5aENF00uVvvlIYwzSkeiKtw/edit?gid=448060195#gid=448060195)) be added separately?
> 
> @natasha-moore-elastic
> 
> I'm having a hard time figuring out how to do this - we just reference out to common schemas which means we can't add a description as the description will differ for each person utilizing it. If you have a good example of how to do this, let me know. I'm just ramping up on OpenAPI.

If we take [PUT exception_lists](https://www.elastic.co/docs/api/doc/kibana/operation/operation-updateexceptionlist#operation-updateexceptionlist) as an example, what we’re aiming to add here is descriptions for request parameters that currently have no description (e.g. `_version`, `description`, `name`, `tags`). If it’s easier, we can reuse the same descriptions we had in [asciidoc docs](https://www.elastic.co/guide/en/security/current/exceptions-api-update-container.html)  (e.g. `description` - Describes the exception container. `name` - The exception container’s name. etc.)
As to how to add those, I’d guess it would be similar to how a description was added for `namespace_type` here: https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/packages/kbn-securitysolution-exceptions-common/api/model/exception_list_common.schema.yaml#L42

yctercero committed:
updating examples

yctercero committed:
Merge branch 'update_exception_list_openapi_info' of github.com:yctercero/kibana into update_exception_list_openapi_info

yctercero committed:
Merge branch 'main' into update_exception_list_openapi_info

yctercero commented:
@vitaliidm @natasha-moore-elastic I think I've addressed all feedback that I can.

In order to add param descriptions I had to not utilize the reference to the common params for generic things like `NonEmptyString` and instead created exception specific ones so we could add a description. To my understanding, when using `$ref: ...` for defining a schema, all other keys are ignored, so even when I set `description:`, it would be ignored. However, we should be good to go now. 

@vitaliidm I updated the PR description to include what @natasha-moore-elastic showed me of how to preview what it will look like in our production mode. 

#### What isn't addressed here

- Issues with query parameters not showing when optional, @natasha-moore-elastic could we make that a separate issue to be addressed? I think this PR still addresses the main goal of getting descriptions and examples ported over

yctercero commented on x-pack/solutions/security/packages/kbn-securitysolution-exceptions-common/api/delete_exception_list/delete_exception_list.schema.yaml:
```diff
@@ -17,25 +17,47 @@ paths:
           description: Either `id` or `list_id` must be specified
           schema:
             $ref: '../model/exception_list_common.schema.yaml#/components/schemas/ExceptionListId'
+          example: 71a9f4b2-c85c-49b4-866f-c71eb9e67da2
```
See comment here - https://github.com/elastic/kibana/pull/205822#issuecomment-2591682113

yctercero commented on x-pack/solutions/security/packages/kbn-securitysolution-exceptions-common/api/find_exception_lists/find_exception_lists.schema.yaml:
```diff
@@ -87,6 +93,30 @@ paths:
+              examples:
+                simpleLists:
+                  value:
+                    data:
```
Fixed!

yctercero commented on x-pack/solutions/security/packages/kbn-securitysolution-exceptions-common/api/read_exception_list_item/read_exception_list_item.schema.yaml:
```diff
@@ -17,25 +17,57 @@ paths:
           description: Either `id` or `item_id` must be specified
           schema:
             $ref: '../model/exception_list_common.schema.yaml#/components/schemas/ExceptionListItemId'
+          example: 71a9f4b2-c85c-49b4-866f-c71eb9e67da2
```
See comment here - https://github.com/elastic/kibana/pull/205822#issuecomment-2591682113

yctercero commented on x-pack/solutions/security/packages/kbn-securitysolution-exceptions-common/api/delete_exception_list_item/delete_exception_list_item.schema.yaml:
```diff
@@ -17,25 +17,57 @@ paths:
           description: Either `id` or `item_id` must be specified
           schema:
             $ref: '../model/exception_list_common.schema.yaml#/components/schemas/ExceptionListItemId'
+          example: 71a9f4b2-c85c-49b4-866f-c71eb9e67da2
```
See comment here - https://github.com/elastic/kibana/pull/205822#issuecomment-2591682113

yctercero commented on x-pack/solutions/security/packages/kbn-securitysolution-exceptions-common/api/read_exception_list_summary/read_exception_list_summary.schema.yaml:
```diff
@@ -17,24 +17,28 @@ paths:
           description: Exception list's identifier generated upon creation
           schema:
             $ref: '../model/exception_list_common.schema.yaml#/components/schemas/ExceptionListId'
+          example: 9e5fc75a-a3da-46c5-96e3-a2ec59c6bb85
```
See comment here - https://github.com/elastic/kibana/pull/205822#issuecomment-2591682113

No these aren't required.

kibanamachine committed:
[CI] Auto-commit changed files from 'yarn openapi:bundle'

kibanamachine committed:
[CI] Auto-commit changed files from 'make api-docs'

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/267226)
* Commit: b4c42e749e84e8305a10e37d17a8f538c87f3811

### Failed CI Steps
* [FTR Configs #22](https://buildkite.com/elastic/kibana-pull-request/builds/267226#019468a0-7aa1-4edc-a5cd-a7c65b1150fd)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/267226#019468a0-7aa1-4edc-a5cd-a7c65b1150fd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/267226/jobs/019468a0-7aa1-4edc-a5cd-a7c65b1150fd/artifacts/019468c1-83a3-40bd-80b2-878a5dc7205d) FTR Configs #<span></span>22 / console app console autocomplete feature Autocomplete behavior JSON autocompletion with placeholder fields

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :yellow_heart: [Build #265773](https://buildkite.com/elastic/kibana-pull-request/builds/265773) was flaky 0be332dd5139a45d4f8ceb654936ba2746d3bb72
* :broken_heart: [Build #264994](https://buildkite.com/elastic/kibana-pull-request/builds/264994) failed bf44ad19994cce68c32caab930e748fe71985883
* :broken_heart: [Build #264971](https://buildkite.com/elastic/kibana-pull-request/builds/264971) failed 917d27b4e026621e94d36571b6ce81ed9ca92aac

cc @yctercero

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/267226","number":267226,"commit":"b4c42e749e84e8305a10e37d17a8f538c87f3811"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265773","number":265773,"commit":"0be332dd5139a45d4f8ceb654936ba2746d3bb72"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264994","number":264994,"commit":"bf44ad19994cce68c32caab930e748fe71985883"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264971","number":264971,"commit":"917d27b4e026621e94d36571b6ce81ed9ca92aac"}],"number":267226}
buildkite-pr-comment-->

natasha-moore-elastic commented:
> @vitaliidm @natasha-moore-elastic I think I've addressed all feedback that I can.
> 
> In order to add param descriptions I had to not utilize the reference to the common params for generic things like `NonEmptyString` and instead created exception specific ones so we could add a description. To my understanding, when using `$ref: ...` for defining a schema, all other keys are ignored, so even when I set `description:`, it would be ignored. However, we should be good to go now.
> 
> @vitaliidm I updated the PR description to include what @natasha-moore-elastic showed me of how to preview what it will look like in our production mode.
> 
> #### What isn't addressed here
> * Issues with query parameters not showing when optional, @natasha-moore-elastic could we make that a separate issue to be addressed? I think this PR still addresses the main goal of getting descriptions and examples ported over

I've checked the preview again and all the body param descriptions look good! We can address the issue with optional query params separately. Thanks!

kibanamachine commented:
Starting backport for target branches: 8.17, 8.x

https://github.com/elastic/kibana/actions/runs/12816849477

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.17|Backport failed because of merge conflicts|
|❌|8.x|Backport failed because of merge conflicts|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 205822
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 205822` locally

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 205822` locally

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 205822` locally

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 205822` locally

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 205822` locally

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 205822` locally

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 205822` locally

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 205822` locally

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 205822` locally

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 205822` locally

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 205822` locally

kibanamachine commented:
Looks like this PR has backport PRs but they still haven't been merged. Please merge them ASAP to keep the branches relatively in sync.

yctercero commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.16|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/209694">](https://github.com/elastic/kibana/pull/209694)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.4 -->

yctercero commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.16|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/209694">](https://github.com/elastic/kibana/pull/209694)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.4 -->
